nixos/doc/manual/development/option-types.section.md

```diff
@@ -35,6 +35,18 @@ merging is handled.
 :   One element of the list *`l`*, e.g. `types.enum [ "left" "right" ]`.
     Multiple definitions cannot be merged.
 
+`types.enumAttrs` *`attrs`*
+
+:   A type that takes one attribute name of the attrset *`attrs`* and maps it
+    to it's value, e.g. `types.enumAttrs { "name" = "value"; ... }`.
+    Multiple definitions cannot be merged.
+
+`types.enumWith` *`mapping`*
```

Comment:
We have somewhat of a pattern where a function named `*With` takes an attribute set for its arguments, to supplement an earlier version of the function that could not be extended because of positional arguments.

I don't think we have a familiar concept of "mapping" in Nix land. "Association list" would be a more accurate term, which I think is worth to be propagated (from Haskell?). Coming to think of it, "name-value list" is even better because we have names for the pair, however, it doesn't have to be a name, right? Isn't the point of this function that it doesn't have to be a string? I think it should be "key-value list" then. Key doesn't have the same hint of stringiness.